src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs

```diff
@@ -60,6 +61,8 @@ private sealed class CategoricalMetaData
 
         protected LightGbmTrainerBase(IHostEnvironment env, LightGbmArguments args, PredictionKind predictionKind, string name)
         {
+            Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");
```

Comment:
>Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US"); [](start = 12, length = 63)

This is nice, and solves all the current problems, but are you sure this is a separate thread we spawn and not are user thread?
I can image users surprise if they code just stop working because somewhere in separate library you setup current thread culture to en-us :)

Reply:
This is current thread at the start of the argument parsing on line 71.

---
In reply to: [199274365](https://github.com/dotnet/machinelearning/pull/454#discussion_r199274365) [](ancestors = 199274365)

Reply:
Reactivating. Just to be clear since it's not obvious, changing a user's culture is not an appropriate solution.

---
In reply to: [199275306](https://github.com/dotnet/machinelearning/pull/454#discussion_r199275306) [](ancestors = 199275306,199274365)

Reply:
ok, let me give you example:
TrainAndPredictIrisModelWithStringLabelTest

Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("fr-FR");
double v = 1.5f;
string t = v.ToString();

put this in begining of test

replace learner with LightGBM and check what you get same 1.5 as parse result

v = double.Parse(t);
pipeline.Add(new LightGbmClassifier());

run training:
            PredictionModel<IrisDataWithStringLabel, IrisPrediction> model = pipeline.Train<IrisDataWithStringLabel, IrisPrediction>();
            v = double.Parse(t);

check value in v


---
In reply to: [199275306](https://github.com/dotnet/machinelearning/pull/454#discussion_r199275306) [](ancestors = 199275306,199274365)

Reply:
proper solution is 
a) change all places where we work with Option argument from ToString(CultureInfo("en-US")) preferably move culture to const variable
b) report this issue to LightGBM folks: [https://github.com/Microsoft/LightGBM/blob/ba3e1ff24fd43ed33d7703881d8efad83a5ed452/include/LightGBM/utils/common.h#L174](https://github.com/Microsoft/LightGBM/blob/ba3e1ff24fd43ed33d7703881d8efad83a5ed452/include/LightGBM/utils/common.h#L174) people who do they own parsing mechanism, have special place in hell.
c) put comments which describe what this is temporary solution and ToString can be reverted to normal state as soon as LightGBM folks fix their parser.

---
In reply to: [199277210](https://github.com/dotnet/machinelearning/pull/454#discussion_r199277210) [](ancestors = 199277210,199275306,199274365)

Reply:
[https://github.com/Microsoft/LightGBM/issues/1481](https://github.com/Microsoft/LightGBM/issues/1481)
here is issue to track

---
In reply to: [199279475](https://github.com/dotnet/machinelearning/pull/454#discussion_r199279475) [](ancestors = 199279475,199277210,199275306,199274365)

Reply:
Thanks for opening the issue on LightGBM repo.

---
In reply to: [199284806](https://github.com/dotnet/machinelearning/pull/454#discussion_r199284806) [](ancestors = 199284806,199279475,199277210,199275306,199274365)